### DISCUSSION TITLE
How to pass arbitrary mysql_options() to the connection

### DISCUSSION THREAD

**Discussion by pvannierop** [PyMySQL/mysqlclient#659] [Q&A]
# Problem
I am trying to pass the `MYSQL_OPT_GET_SERVER_PUBLIC_KEY` option to the connection. `MYSQL_OPT_GET_SERVER_PUBLIC_KEY`  is one of the options described in [_mysql_options()_](https://dev.mysql.com/doc/c-api/8.0/en/mysql-options.html). I am looking for a way to achieve this.

# Question
How are arbitrary _mysql_options() that are not described as parameters to the `MySQLdb.connect(...)` method?

# Docs
The docs state the following:

```
connect(parameters…)
   Constructor for creating a connection to the database. Returns a Connection Object. Parameters are the same as for the 
   MySQL C API. In addition, there are a few additional keywords that correspond to what you would pass mysql_options() 
   before connecting. Note that some parameters must be specified as keyword arguments!
```

At the _MySQL C API function mapping_ section (see [here](https://mysqlclient.readthedocs.io/user_guide.html#mysql-c-api-function-mapping)) it states:

|  C API   |  _mysql   |
|---- |----- |
|  mysql_options() | various options to _mysql.connect() |

I think that the docs are a bit unclear on whether it is possible to pass _mysql_options()_ to the connection?

Any help would be appreciated greatly.

**Comment by methane** [comment]
No way. Use `~/my.cnf`.
